build-scripts/generateServiceClient.ts

```diff
@@ -0,0 +1,197 @@
+/*!
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as child_process from 'child_process'
+import * as del from 'del'
+import * as fs from 'fs'
+import * as os from 'os'
+import * as path from 'path'
+
+/**
+ * This script uses the AWS JS SDK to generate service clients where the client definition is contained within
+ * this repo. Client definitions are added at the bottom of this script.
+ */
+
+interface ServiceClientDefinition {
+    serviceName: string
+    serviceJsonPath: string
+}
+
+async function generateServiceClients(serviceClientDefinitions: ServiceClientDefinition[]): Promise<void> {
+    const tempJsSdkPath = fs.mkdtempSync(path.join(os.tmpdir(), 'vsctk-generate'))
+    console.log(`Temp JS SDK Repo location: ${tempJsSdkPath}`)
+    console.log('Serivce Clients to Generate: ', serviceClientDefinitions.map(x => x.serviceName).join(', '))
+
+    try {
+        await cloneJsSdk(tempJsSdkPath)
+
+        await insertServiceClientsIntoJsSdk(tempJsSdkPath, serviceClientDefinitions)
+
+        await runTypingsGenerator(tempJsSdkPath)
+
+        await integrateServiceClients(tempJsSdkPath, serviceClientDefinitions)
+
+        console.log('Done generating service client(s)')
+    } finally {
+        // Clean up the temp path
+        del.sync([tempJsSdkPath], { force: true })
+    }
+}
+
+async function cloneJsSdk(destinationPath: string): Promise<void> {
+    console.log('Cloning AWS JS SDK...')
+
+    // Output stderr while it clones so it doesn't look frozen
+    return new Promise<void>((resolve, reject) => {
+        const exec = child_process.exec(
+            `git clone --depth 1 https://github.com/aws/aws-sdk-js.git ${destinationPath}`,
```

Comment:
Unless we intentionally want to invoke a shell (almost never), it's a good practice to use the "no shell" variant of such APIs. In this case it looks like `child_process.execFile` avoids the shell.

https://nodejs.org/api/child_process.html#child_process_child_process_execfile_file_args_options_callback

Generally when args are specified as a single string that's a clue that it's invoking a shell. That means that quoting issues are platform-specific. By using a "list" form of args, one avoids the shell, and quoting issues are avoided.

So instead this would look like:

    child_process.execFile('git', 'clone', '--depth', '1', ...)